File: src/tb-detection/repository.py
import repoStrategy as rs

class Repository():
    def __init__(self, strategy_name):
        self.data = {}
        self.strategy = rs.RepoStrategyFactory.get_strategy(strategy_name)

    def add_to_data(self, marker_id, data):
        self.data.setdefault(str(marker_id), data).update()

    def setup(self):
        self.strategy.setup()

    # Update the UDP thread with the data
    def send_data(self):
        self.strategy.set_data(self.data)
        self.data = {}

    def check_for_terminate(self):
        return self.strategy.terminate
        
    def update(self, marker_json, marker_id):
        self.add_to_data(marker_id, marker_json)

File: src/detector/DetectorTest/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DetectorTest
{
    internal class Repository
    {
        private Dictionary<string, object> data = new Dictionary<string, object>();
        private IRepoStrategy strategy;

        public Repository(string strategyName)
        {
            strategy = RepoStrategyFactory.GetStrategy(strategyName);
        }

        public void AddToData(int markerId, object data)
        {
            string markerIdStr = markerId.ToString();
            if (this.data.ContainsKey(markerIdStr))
            {
                this.data[markerIdStr] = data;
            }
            else
            {
                this.data.Add(markerIdStr, data);
            }
        }

        public void Setup()
        {
            strategy.Setup();
        }

        public void CloseThreads()
        {
            if (!strategy.Terminate)
            {
                strategy.Terminate = true;
            }
        }

        public void SendData()
        {
            strategy.SetData(data);
            data.Clear();
        }

        public bool CheckForTerminate()
        {
            return strategy.Terminate;
        }

        public void Update(Dictionary<string, object> markerJson, int markerId)
        {
            AddToData(markerId, markerJson);
        }
    }

}
